tests/scale/test_soak.py

```diff
@@ -0,0 +1,148 @@
+"""
+import pprint
+import logging
+import time
+from threading import Thread, Lock
+from typing import List, Set
+from xml.etree import ElementTree
+
+import config
+import jenkins
+import jenkins_common
+import pytest
+import sdk_dcos
+import sdk_marathon
+import sdk_quota
+import sdk_security
+import sdk_utils
+import shakedown
+import json
+
+from sdk_dcos import DCOS_SECURITY
+
+log = logging.getLogger(__name__)
+
+@pytest.mark.soak
+def test_scaling_soak(job_count,
+                      single_use: bool,
+                      run_delay,
+                      work_duration,
+                      external_volume: bool,
+                      scenario,
+                      pinned_hostname,
+                      pinned_host_volume,
+                      datadog_api_key,
+                      datadog_plugin_hostname) -> None:
+
+    """Launch a soak test scenario.
+
+    Configuration, installation and jobs are launched serially.
+
+    Args:
+        job_count: Number of Jobs on each Jenkins master
+        single_use: Mesos Single-Use Agent on (true) or off (false)
+        run_delay: Jobs should run every X minute(s)
+        work_duration: Time, in seconds, for generated jobs to sleep
+        scenario: Jenkins senarios, one of 'sleep' or 'buildmarathon'
+        pinned_hostname: IP address of node to pin Jenkins to.
+        pinned_host_volume: Location of temporary storage on specified host.
+        datadog_api_key: API required for DataDog metrics service.
+        datadog_plugin_hostname: Hostname reported to the DataDog metrics service.
+    """
+    security_mode = sdk_dcos.get_security_mode()
+    
+    # create marathon client
+    marathon_client = shakedown.marathon.create_client()
+
+    service_name = "jenkins-soak"
+    
+    # create service accounts
+    sdk_security.install_enterprise_cli()
+
+    # create service accounts
+    jenkins_common.create_service_accounts(service_name, security=security_mode)
+
+    # launch Jenkins services
+    jenkins_common.install_jenkins(service_name,
+                                    client=marathon_client,
+                                    external_volume=external_volume,
+                                    security=security_mode,
+                                    pinned_hostname=pinned_hostname,
+                                    pinned_host_volume=pinned_host_volume)
+   
+    # install DataDog metrics plugin
+    jenkins_common.install_jenkins_datadog_metrics_plugin(service_name, datadog_plugin_hostname, datadog_api_key)
+    
+    # the rest of the commands require a running Jenkins instance
+    jenkins_common.create_jobs(service_name,
+                                jobs=job_count,
+                                single=single_use,
+                                delay=run_delay,
+                                duration=work_duration,
+                                scenario=scenario)
+
+@pytest.mark.soakcleanup
+def test_cleanup_soak(mom) -> None:
```

Comment:
this method can also clean up any created service accounts

Reply:
Seems todays data services dont ever delete the service accounts they create, since they will reuse them, so this is not a blocker

Reply:
Fixed this under jenkins_common.cleanup_jenkins_install(), we delete the corresponding service_account.